community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreAbstractNodeCursor.java

```diff
@@ -140,16 +146,29 @@ public long id()
         return singleLabelCursor.get().init( parseLabelsField( nodeRecord ).get( nodeStore ), labelId );
     }
 
+    private Lock shortLivedReadLock()
+    {
+        Lock lock = lockService.acquireNodeLock( nodeRecord.getId(), LockService.LockType.READ_LOCK );
+        if ( lockService != NO_LOCK_SERVICE )
+        {
+            // It's safer to re-read the node record here, specifically nextProp, after acquiring the lock
+            nodeStore.loadRecord( nodeRecord.getId(), nodeRecord );
```

Comment:
`loadRecord` doesn't throw `InvalidRecordException` actually. Although it may throw `UnderlyingStorageException` if there's an `IOException` from the page cache. Perhaps we should have a `finally` thing here.